apis/cqn.d.ts

```diff
@@ -15,10 +15,14 @@ export type SELECT = { SELECT: {
   columns?: column_expr[],
   excluding?: string[],
   where?: predicate,
-  having?: predicate,
   groupBy?: expr[],
+  having?: predicate,
   orderBy?: ordering_term[],
   limit?: { rows: val, offset: val },
+  forUpdate?: { wait: number },
+  forShareLock?: { wait: number },
+  search?: predicate,
```

Comment:
On an `SELECT` a `search` property may exist, this is correct. Can be seen in our tests, e.g. here:

https://github.com/cap-js/cds-dbs/blob/732a2f385074f50b87ff9715b8bdf48d28a36309/db-service/test/cqn4sql/search.test.js#L28